### ISSUE TITLE
Closing bracket not detected

### ISSUE THREAD

**Issue by pecigonzalo** [mauve/vscode-terraform#89]
As you can see in this screenshot closing `}` is no detected as a punctuation/sign
![image](https://user-images.githubusercontent.com/9908105/40783609-479e4bc4-64e3-11e8-8c2c-fec326aef345.png)

Opening `{` seems to work.
![image](https://user-images.githubusercontent.com/9908105/40783640-5fc4c34a-64e3-11e8-86f9-feb02c492abd.png)

**Comment by mauve** [comment]
I think the grammar file (https://github.com/mauve/vscode-terraform/blob/master/syntaxes/terraform.json) is full of improvement-abilities :) the syntax of that file baffles me though so if you know how it works help is greatly appreciated.

I have some other changes I am currently working on and I have visitors so I will not be able to work on this for at least a week.

**Comment by pecigonzalo** [comment]
Yep, I was trying to search through the file and understand it, but I cant find any docs. Didnt you write that file?

**Comment by mauve** [comment]
There is almost no documentation on that file-format. I didn't write the file from scratch, I kinda took several different `tmLanguage` syntax files which I found on the internet and took different parts from different files.

The file format is `tmLanguage` which has two syntaxes, one JSON one and one custom one. I used the custom one before, but migrated to JSON a couple of months back.

Here are some docs: https://manual.macromates.com/en/language_grammars
